authorization/src/main/java/com/terabite/authorization/config/SecurityConfig.java

```diff
@@ -23,53 +26,83 @@
-        return new BCryptPasswordEncoder();
-    }
+        @Bean
+        public AuthenticationProvider authenticationProvider(@Lazy final LoginService loginService) {
+                DaoAuthenticationProvider authenticationProvider = new DaoAuthenticationProvider();
+                authenticationProvider.setUserDetailsService(loginService);
+                authenticationProvider.setPasswordEncoder(passwordEncoder());
+                return authenticationProvider;
+        }
 
-    @Bean
-    public UserDetailsService userDetailsService() {
-        return new LoginService();
-    }
+        @Bean
+        public AuthenticationManager authenticationManager(AuthenticationConfiguration authenticationConfiguration)
+                        throws Exception {
+                return authenticationConfiguration.getAuthenticationManager();
+        }
 
-    @Bean
-    public AuthenticationProvider authenticationProvider() {
-        DaoAuthenticationProvider authenticationProvider = new DaoAuthenticationProvider();
-        authenticationProvider.setUserDetailsService(userDetailsService());
-        authenticationProvider.setPasswordEncoder(passwordEncoder());
-        return authenticationProvider;
-    }
+        // Allows all users for all routes.
+        // CSRF disabled because it breaks authorization since backend is stateless
+        @Bean
+        public SecurityFilterChain filterChain(HttpSecurity http, @Lazy final AuthenticationProvider authProvider)
+                        throws Exception {
+                http.csrf((csrf) -> csrf.disable())
+                                .exceptionHandling((exceptionHandler) -> exceptionHandler
+                                                .accessDeniedHandler(new CustomAccessDeniedHandler()))
+                                .authorizeHttpRequests((authorizeHttpRequests) -> {
+                                        authorizeHttpRequests
+                                                        .dispatcherTypeMatchers(DispatcherType.ERROR,
+                                                                        DispatcherType.FORWARD)
+                                                        .permitAll();
+                                        for (String route : AUTHENTICATED_ROUTES) {
+                                                authorizeHttpRequests
+                                                                .requestMatchers(String.format("/%s%s", VERSION_PREFIX,
+                                                                                route))
+                                                                .authenticated();
+                                        }
 
-    @Bean
-    public AuthenticationManager authenticationManager(AuthenticationConfiguration authenticationConfiguration) throws Exception {
-        return authenticationConfiguration.getAuthenticationManager();
-    }
+                                        for (String route : PERMIT_ALL_ROUTES) {
```

Comment:
Not sure if having an explicit "whitelist" is the best approach here. Having both a whitelist (`PERMIT_ALL_ROUTES`) and blacklist (`AUTHENTICATED_ROUTES`) seems excessive, since routes not in either are always blocked anyways. Just having a blacklist would be easier to understand and modify for other people to restrict their routes, for the same effect.

Something like would control authorization while allowing all other endpoints
```java
for (String r : blacklist) {
    authorizeHttpRequests.requestMatchers("...").authenticated();
}
authorizeHttpRequests.anyRequest().permitall();
```

Reply:
Ok, I saw that. I was not sure how this worked exactly, so I thought doing the anyRequest.permitAll invalidated the authentication requirements.. This is much cleaner, I will change to this solution.